sklearn/preprocessing/_polynomial.py

```diff
@@ -1027,18 +1009,10 @@ def transform(self, X):
         n_splines = self.bsplines_[0].c.shape[1]
         degree = self.degree
 
-        # TODO: Remove this condition, once scipy 1.10 is the minimum version.
-        #       Only scipy >= 1.10 supports design_matrix(.., extrapolate=..).
-        #       The default (implicit in scipy < 1.10) is extrapolate=False.
-        scipy_1_10 = sp_version >= parse_version("1.10.0")
         # Note: self.bsplines_[0].extrapolate is True for extrapolation in
         # ["periodic", "continue"]
-        if scipy_1_10:
-            use_sparse = self.sparse_output
-            kwargs_extrapolate = {"extrapolate": self.bsplines_[0].extrapolate}
-        else:
-            use_sparse = self.sparse_output and not self.bsplines_[0].extrapolate
-            kwargs_extrapolate = dict()
+        use_sparse = self.sparse_output
+        kwargs_extrapolate = {"extrapolate": self.bsplines_[0].extrapolate}
```

Comment:
I would use `extrapolate=self.bsplines_[0].extrapolate` in the call where `kwargs_extrapolate` is used.

Similar thing as previous comment, `kwargs_extrapolate` was to cover more easily scipy < 1.10 and scipy >= 1.10